backend/src/email/middlewares/email-template.middleware.ts

```diff
@@ -99,49 +135,49 @@ const uploadCompleteHandler = async (req: Request, res: Response): Promise<Respo
     if (emailTemplate === null){
       throw new Error('Template does not exist, please create a template')
     }
-
-    // Updates metadata in project
-    await CampaignService.updateCampaignS3Metadata({ key: s3Key, campaignId, filename })
-
+  
     // carry out templating / hydration
     // - download from s3
-
-    const { records, hydratedRecord } = await EmailTemplateService.client.testHydration({
-      campaignId: +campaignId,
-      s3Key,
-      templateSubject: emailTemplate.subject,
-      templateBody: emailTemplate.body as string,
-      templateParams: emailTemplate.params as string[],
-    })
-
-    if (EmailTemplateService.hasInvalidEmailRecipient(records)) throw new InvalidRecipientError()
-
-    const recipientCount: number = records.length
-
-    // START populate template
-    logger.info(`before email.addToMessageLogs; campaignId=${campaignId}`)
-    await EmailTemplateService.addToMessageLogs(+campaignId, records)
-    logger.info(`after email.addToMessageLogs; campaignId=${campaignId}`)
-
-    if (!res.headersSent){
-      return res.json({
-        'num_recipients': recipientCount,
-        preview: {
-          ...hydratedRecord,
-          // eslint-disable-next-line @typescript-eslint/camelcase
-          reply_to: emailTemplate.replyTo || null,
-        },
+    try {
+      const { records, hydratedRecord } = await EmailTemplateService.client.testHydration({
+        campaignId: +campaignId,
+        s3Key,
+        templateSubject: emailTemplate.subject,
+        templateBody: emailTemplate.body as string,
+        templateParams: emailTemplate.params as string[],
       })
+  
+      if (EmailTemplateService.hasInvalidEmailRecipient(records)) throw new InvalidRecipientError()
+      
+      const recipientCount: number = records.length
+
+      await updateCampaignAndMessages(s3Key, campaignId, filename, records)
+
+      if (!res.headersSent) {
+        return res.json({
+          'num_recipients': recipientCount,
+          preview: {
+            ...hydratedRecord,
+            // eslint-disable-next-line @typescript-eslint/camelcase
+            reply_to: emailTemplate.replyTo || null,
+          },
+        })
+      } 
+    } catch(err) {
+      logger.error(`Error parsing file for campaign ${campaignId}. ${err.stack}`)
+      throw err
     }
-
   } catch (err) {
     if (!res.headersSent){
-      return res.status(400).json({ message: err.message })
+      if (err instanceof RecipientColumnMissing || err instanceof MissingTemplateKeysError || err instanceof InvalidRecipientError) {
```

Comment:
@chinying I noticed that in #321 we changed the function such that we no longer return `next(err)`  and returned every error as a 400. Does that apply to sms too or only for email? I changed it back so that it is easier to tell the difference.